src/types/Student.ts

```diff
@@ -0,0 +1,16 @@
+import {Relationship} from "@/types/Relationship"
+
+export type AttributeMap = {
+    [attribute: number]: number[]
```

Comment:
nit;

There are built in types for things like this, I think it's Record for this and the one below (instead of giving a name to the key)

Reply:
Ok, nice. I'm still a bit new to TS and I figured there was better way, but couldn't find it

Reply:
Nah no worries, the utility types in typescript are kinda fire, skim them at some point for big brains